models/cls_model.py

```diff
@@ -0,0 +1,284 @@
+from models.training_model_api import TrainingModel
+from models.preprocessing import get_iob_entity_encoding, cls_preprocess, text_split_preprocess, df_to_dataloader, glob_to_snapshot
+from models.preprocessing import entity_labels
+from transformers import AutoTokenizer, BertForSequenceClassification
+from sklearn.metrics import accuracy_score, confusion_matrix
+from seqeval.metrics import classification_report
+import torch
+from models.utils import generate_default_parameters
+
+
+class ClsModel(TrainingModel):
+    """
+    Model for training on Clasification task
+    """
+
+    def __init__(self):
+        super().__init__()
+        self.entity_labels = entity_labels
+
+    def set_parameters(self, parameters):
+        """Set parameters dictionary and setup model, tokenizer, and optimizer
+
+        Parameters
+        ----------
+        parameters : dict
+            parameters dictionary, model.__getattr__ will try to look here if the name doesn't exist in self.__dict__
+        """
+        super().set_parameters(parameters)
+        self.parameters['num_labels'] = 5
+
+        # Check if we should use cuda
+        if self.use_cuda and torch.cuda.is_available():
+            self.device = 'cuda'
+        else:
+            self.device = 'cpu'
+        # Setup model bits
+        self.tokenizer = AutoTokenizer.from_pretrained(self.tokenizer_url)
+        self.nn = self._init_nn()
+        self.optimizer = torch.optim.Adam(params=self.nn.parameters(), lr=self.learning_rate)
+
+    def _init_nn(self):
+        """Initialize the PyTorch BERT nn and put onto device"""
+        self.nn = BertForSequenceClassification.from_pretrained(self.base_model_url, num_labels=self.num_labels)
+        self.nn.to(self.device)
+        return self.nn
+
+    def preprocess(self, data_snapshot, generate_labels=True):
+        """Transform data snapshot into a dataframe
+        Returns a dataframe of the text without labels in the case of inference when generate_labels is set to False
+
+        Parameters
+        ----------
+        data_snapshot : dict of Documents
+            data snapshot for training/validation/test data
+        generate_labels : bool, optional
+            Whether to generate training labels (for training and validation), by default True
+
+        Returns
+        -------
+        pd.DataFrame
+            Data transformed into dataframe with keys for ['text', 'id'] and ['label'] if labels are generated
+        """
+        # if generate_labels:
```

Comment:
Keep the logic for generate labels because if text is passed without labels present (e.g. in inference) then we should do some other preprocessing (likely just text splitting).